### PR TITLE
Updates to use OK for key parsing

### PR DESCRIPTION
**Pull request by e0d**
@feanil 
@singingwolfboy 

Thoughts?

### CODE REVIEW

**Comment by singingwolfboy** [comment]
Looks good to me. :+1:

**Comment by stvstnfrd** [comment]
Hey @e0d, are you all running this version of the certs repo? Last I heard, you were using a different version.

**Comment by e0d** [comment]
@stvstnfrd We're not yet, but I was sad that this didn't support the new course key format, there was some chatter about it on the google group.

**Comment by e0d** [comment]
@stvstnfrd Please review.  I've needed up upgrade reportlab as the version in requirement.txt referenced a version of pillow that is no longer available.  The dependency seems to have been renamed from pillow to Pillow.

**Comment by stvstnfrd** [comment]
Gotcha @e0d , saw the post too.

We've been using this repo for a long time now (we were core committers when @jrbl was still here and had commit access to this repo). Do you mind if I test this out with our config/deploy before merging?

**Comment by stvstnfrd** [comment]
Ah, didn't see your follow-up; thanks, I'm spinning this up.

**Comment by e0d** [comment]
@stvstnfrd let me know if you have or do encounter any issues.

**Comment by stvstnfrd** [comment]
@e0d I was able to successfully generate a cert using a slash-separated key;
I have _not_ yet tested this using an OpaqueKey (we're only running
certs for courses w/ SSKs).

So we'll want to add test coverage for opaque keys. This _should_ be as
easy as adding a relevant entry to `cert-data.yml`;
`tests/gen_cert_test.py` iterates over every key in the YAML file and
uses it to attempt generating a cert.

The following locations still reference slash-separated keys; will you
update these too?
- line 195: gen_cert.py
- line  58: README.md
- line  64: README.md

File: gen_cert.py [hunk 1]
```diff
@@ -233,12 +236,16 @@ def interstitial_factory():
...
-        tmp_org, tmp_course, tmp_run = course_id.split('/')
-        self.course = cert_data.get('COURSE', tmp_course)
-        self.org = cert_data.get('ORG', tmp_org)
+        # if COURSE or ORG is set in the configuration attempt to parse using
+        # new style course_keys and fallback to using from_deprecated_string
+        # upon failure
+        try:
+            course_key = CourseKey.from_string(course_id)
+        except InvalidKeyError:
+            course_key = SlashSeparatedCourseKey.from_deprecated_string(course_id)
```
Comment (caijim) [inline gen_cert.py]: why not just directly use CourseLocator.from_string ?

Comment (e0d) [inline gen_cert.py]: @ormsbee Do we have a preferred idiom?

Comment (ormsbee) [inline gen_cert.py]: Please use `CourseKey.from_string()` -- that should handle both the new style locators and the old slash separated course key cases.

**Comment by e0d** [comment]
@stvstnfrd 

Since this change will support both style keys, do you think the best thing is to provide examples of both in the readme?

**Comment by stvstnfrd** [comment]
@e0d We _could_, though I generally avoid documenting deprecated
functionality.

In this case, anyone currently using this code _knows_ how to use it
with SSKs, since that's all that was supported. For those people, things
will keep working as before (and they probably aren't reading the README much).

For new adopters, they're probably already running Split since it's the
new default, so I'd rather not distract them w/ "old" instructions.

And really, since the code supports all course key formats, it should just work
with anything that's thrown at it, so I'm cool w/ only showcasing the "preferred",
new format.

**Comment by e0d** [comment]
@stvstnfrd @caijim followup review?

**Comment by stvstnfrd** [comment]
Thanks for updating the docs and tests @e0d !

+1 to squash and merge

**Pull request merged**